tools/2024-06-gitops-feature-spec.md

```diff
@@ -0,0 +1,345 @@
+2. Then I push the changes to my git repository. 
+3. This triggers Flux/ArgoCD to detect the change and update the `frontend` resource with the new function. I use Flux/ArgoCD to monitor the health status of the Kubernetes cluster and immediately realize the pod associated with my `frontend` resource is failing.
+4. I review my code change again and realize I created a bug in the function which is causing the resource to fail. 
+5. To resolve, I:
+   1. Use the Flux/ArgoCD rollback commands to restart my `frontend` resource pods with the previous code version so as not to disrupt my customers. 
+   1. Proceed with a new code change to my `helper.ts` that resolves the bug, then push this change to my repository. 
+6. Flux/ArgoCD now deploys my bug fix and I can see in Flux/ArgoCD that the pod has restarted in a healthy state with my new code. 
+
+> As a Developer, I'm very happy with this experience because I get the cool new features of Radius (like self-serve infrastructure deployment via Recipes and the App Graph/Dashboard) in addition to the features I already know and love in Flux, like health monitoring and rollback.
+
+__Requirements resulting from these scenarios:__ 
+- Flux/ArgoCD must be able to detect the file change described above and must be able to read the env.bicep file as required to deploy the Radius app correctly 
+- Radius types and Bicep files must be able to be read by Flux/ArgoCD which currently does not have that ability.
+- Flux/ArgoCD must be able to detect Radius resource types and execute Flux/ArgoCD commands against those types, including rollback *list of any other required Flux commands*.
+- Based on user feedback this could be just testing that commands such as dry run mechanisms work
+
+## Key investments
+<!-- List the features required to enable this scenario. -->
+
+### Feature 1: GitOps tooling can detect and understand Radius definition files
+<!-- One or two sentence summary -->
+GitOps tools (beginning with Flux and ArgoCD) must be able to detect the file change in the git repo described above and must be able to read the `env.bicep` file as required to deploy the Radius app correctly.
+
+GitOps tools (beginning with Flux and ArgoCD) must be able to detect changes to Radius applications and other resource type definitions in the git repo and execute the appropriate deployment, etc. commands against those types, including rollback operations.
+
+### Feature 2: GitOps tooling can execute Radius commands
+<!-- One or two sentence summary -->
+Once Radius definition file changes are detected and understood, GitOps tools (beginning with Flux and ArgoCD) must be able to execute deployments for the Radius resources defined in the git repo.
+
+### Feature 3: GitOps tooling can rollback Radius resources
+<!-- One or two sentence summary -->
+GitOps tools (beginning with Flux and ArgoCD) must be able to execute rollback operations for Radius resources defined in the git repo.
+
```

Comment:
Suggestion: `Feature 4: Declarative Deletion Must Work`

If I delete a resource in an application from my bicep file and push it, that resource should be deleted. This is not a behavior of bicep today but it's important for this usecase.